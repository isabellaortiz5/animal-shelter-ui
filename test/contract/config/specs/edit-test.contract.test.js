import { provider } from '../init-pact';
import { Matchers } from '@pact-foundation/pact';
import { AnimalController } from '../../../../controllers';

describe('Animal Service', () => {
    describe('When a request to edit an animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                uponReceiving: 'a request to edit an animal',
                state: "update animal",
                withRequest: {
                    method: 'PUT',
                    path: '/animals/Joy',
                },
                willRespondWith: {
                    status: 200,
                    body:{
                        name: Matchers.like('Joy'),
                        breed: Matchers.like("Bengali"),
                        gender: Matchers.like("Female"),
                        vaccinated: Matchers.boolean(true)
                    }
                }
            });
        });
    
        it('Animal edited', async () => {
            const response = await AnimalController.updateAnimal("Joy");
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(() => provider.finalize());

    })
});
