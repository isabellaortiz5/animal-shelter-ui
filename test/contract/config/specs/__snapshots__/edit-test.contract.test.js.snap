// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to edit an animal is made Animal edited 1`] = `
Object {
  "breed": "Bengali",
  "gender": "Female",
  "name": "Joy",
  "vaccinated": true,
}
`;
